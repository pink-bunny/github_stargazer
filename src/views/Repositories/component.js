import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Button } from 'react-native-elements';
import { useNavigation } from '@react-navigation/native';

import Header from '../shared/Header';
import Empty from './Empty';
import List from './List';

const Repositories = ({ repositories }) => {
  const navigation = useNavigation();

  return (
    <>
      <Header
        centerTitle="Github Stargather"
        rightComponent={(
          <Button
            testID="ctaAddRepository"
            onPress={() => navigation.navigate('AddRepository')}
            buttonStyle={{ padding: 12 }}
            icon={(
              <Icon
                name="plus"
                type="octicon"
                size={18}
              />
            )}
            type="clear"
          />
        )}
      />

      <List repositories={repositories} />
    </>
  );
};

Repositories.defaultProps = {
  repositories: [],
};

Repositories.propTypes = {
  repositories: PropTypes.arrayOf(PropTypes.shape()),
};

export default Repositories;
