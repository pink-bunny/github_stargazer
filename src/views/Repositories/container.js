import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { repositoriesSelector } from '../../state/concepts/repository/selectors';
import RepositoriesComponent from './component';

// eslint-disable-next-line react/prefer-stateless-function
class ListContainer extends Component {
  render() {
    const { repositories } = this.props;

    return <RepositoriesComponent repositories={repositories} />;
  }
}

ListContainer.defaultProps = {
  repositories: null,
};

ListContainer.propTypes = {
  repositories: PropTypes.arrayOf(PropTypes.shape()),
};

const mapStateToProps = (state) => ({
  repositories: repositoriesSelector(state),
});

export default connect(mapStateToProps)(ListContainer);
