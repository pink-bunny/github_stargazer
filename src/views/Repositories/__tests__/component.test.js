import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Repositories from '../component';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

describe('Repositories component.', () => {
  it('Opening Add repository screen', () => {
    const { queryByTestId } = render(<Repositories />);

    fireEvent.press(queryByTestId('ctaAddRepository'));
    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith('AddRepository');
  });

  it('Empty screen', () => {
    const { queryByTestId, toJSON } = render(<Repositories />);

    expect(queryByTestId('emptyView')).toBeEnabled();
    expect(queryByTestId('emptyViewText')).toHaveTextContent('No repositories added');
    expect(toJSON()).toMatchSnapshot();

    expect(queryByTestId('repositories')).toBeNull();
  });

  it('Repositories screen', () => {
    const props = {
      repositories: [{ id: 0 }, { id: 1 }],
    };
    const { queryAllByTestId, toJSON } = render(<Repositories {...props} />);

    expect(queryAllByTestId('repositoryItem')).toHaveLength(2);
    expect(toJSON()).toMatchSnapshot();
  });
});

