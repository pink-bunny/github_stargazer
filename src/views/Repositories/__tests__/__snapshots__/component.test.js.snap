// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories component. Empty screen 1`] = `
Array [
  <View
    centerTitle="Github Stargather"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
        "position": "relative",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
        "zIndex": 1,
      }
    }
    testID="headerContainer"
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        testID="headerTitle"
      >
        Github Stargather
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ctaAddRepository"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="emptyView"
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#242424",
                  "fontSize": 28,
                },
                Object {
                  "backgroundColor": "transparent",
                  "marginBottom": 16,
                },
                Object {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      testID="emptyViewText"
    >
      No repositories added
    </Text>
  </View>,
]
`;

exports[`Repositories component. Repositories screen 1`] = `
Array [
  <View
    centerTitle="Github Stargather"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
        "position": "relative",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
        "zIndex": 1,
      }
    }
    testID="headerContainer"
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        testID="headerTitle"
      >
        Github Stargather
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ctaAddRepository"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    testID="repositories"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "#c5c5c5",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
          },
        ]
      }
      testID="repositoryItem"
    >
      <Text />
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "#c5c5c5",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
          },
        ]
      }
      testID="repositoryItem"
    >
      <Text />
      <Text />
    </View>
  </View>,
]
`;
